src/ssh_agent.rs

```diff
@@ -233,11 +267,17 @@ impl<
             }
         };
 
+
         match key.private_key() {
             Some(private_key) => {
                 writebuf.push(msg::SIGN_RESPONSE);
-                let signature = private_key
-                    .try_sign(data)
+                let result: Result<Signature,_> = if let KeypairData::Rsa(keypair) = private_key.key_data() {
+                    let alg = flags_enum.get_hash_alg();
+                    (keypair,alg).try_sign(data)
```

Comment:
```suggestion
                    // Sha1 is deprecated but does not need to be filtered out since the sha1 feature is disabled for the ssh key crate
                    (keypair,alg).try_sign(data)
```